ml/jml/feature_set.cc

```diff
@@ -94,9 +94,9 @@ replace(const Feature & feat, float val)
             it->second = val;
             ++it;
             int start = it - features.begin();
-            int pos = 0;
+            int pos = start;
```

Comment:
This should go in separately as a separate PR and issue; it's a pure bug fix that should not depend on the rest.